src/main.cpp

```diff
@@ -1590,6 +1590,18 @@ static void AlertNotify(const std::string& strMessage, bool fThread)
         runCommand(strCmd);
 }
 
+static void AlertNotifyOnce(const std::string& strMessage, bool fThread)
+{
+    static bool fWarned = false;
+
+    if (!fWarned) {
+      // strMiscWarning is read by GetWarnings(), called by Qt and the JSON-RPC code to warn the user
+      strMiscWarning = strMessage;
```

Comment:
What about moving the setting of strMiscWarning to AlertNotify?
All uses of AlertNotify set strMiscWarning except one in CheckForkWarningConditions, and I think that's a mistake.